internal/testutil/tempdir.go

```diff
@@ -72,3 +76,11 @@ func (td *TempDir) Write(file string, content []byte) *TempDir {
 	}
 	return td
 }
+
+func (td *TempDir) WritePlugin(file string, plugin index.Plugin) *TempDir {
```

Comment:
I figure this will be useful if we want to write more tests that involve the filesystem

Reply:
eh, it's ok. i suspect we might forget to use it. :)

Reply:
pls add `td.t.Helper()` and it's lgtm

Reply:
Done.